rust/src/transformations/make_stable_expr/expr_str/expr_strptime/mod.rs

```diff
@@ -0,0 +1,107 @@
+use polars::prelude::*;
+use polars_plan::dsl::Expr;
+
+use crate::core::{Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, OuterMetric, SeriesDomain};
+use crate::error::*;
+use crate::polars::{literal_value_of, ExprFunction};
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that returns a `to_date` expression.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The str.strptime expression
+pub fn make_expr_strptime<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Function {
+        input: inputs,
+        function: FunctionExpr::StringExpr(StringFunction::Strptime(to_type, mut strptime_options)),
+        ..
+    } = expr
+    else {
+        return fallible!(MakeTransformation, "expected str.strptime expression");
+    };
+
+    let Ok([input, ambiguous]) = <&[_; 2]>::try_from(inputs.as_slice()) else {
+        return fallible!(
+            MakeTransformation,
+            "str.strptime must have two arguments, found {}",
+            inputs.len()
+        );
+    };
+
+    if !matches!(
+        to_type,
+        DataType::Time | DataType::Datetime(_, _) | DataType::Date
+    ) {
+        return fallible!(
+            MakeTransformation,
+            "str.strptime dtype must be Time, Datetime or Date, found {}",
+            to_type
+        );
+    }
+    // Strict casting makes the transformation unstable: errors tell you things about private data.
+    // Could throw an error if strict, but it is the default, so for ease-of-use it is forced to be non-strict.
+    // It is also ok for overflow to wraparound.
```

Comment:
What is the range of dates which we can represent? If dates earlier than 1970 wrap around, that is going to be undesirable in many cases.

Reply:
Dates and datetimes are signed offsets from the epoch, so dates earlier than 1970 don't wrap until much earlier time points.